### PR TITLE
Delete User from store

### PR DESCRIPTION
**Pull request by kjetilk**
Here's my first stab at deleting user from the store. This is implemented as part of https://github.com/solid/node-solid-server/issues/177

The test doesn't pass, and I'm not sure the test is OK, but I pushing makes it easier to discuss.

### CODE REVIEW

File: src/user-store.js [hunk 1]
```diff
@@ -194,6 +194,20 @@ class UserStore {
...
+   * @param user {UserAccount}
+   *
+   * @return {Promise}
+   */
+
+  deleteUser (user) {
+    let userKey = UserStore.normalizeIdKey(user.id)
+
+    return Promise.resolve()
+      .then(() => this.backend.del('users', userKey))
```
Comment (megoth) [inline src/user-store.js]: I would prefer `this.backend.remove` as that is what in exposed in the documentation for https://github.com/solid/kvplus-files, but I see that there is an alias with `del` there, so I'm fine with it.

Comment (megoth) [reply src/user-store.js]: Just got to know that `del` is recommended by @dmitrizagidulin , so ignore the comment above ^_^

**Comment by dmitrizagidulin** [comment]
@megoth ahhh no no, sorry bout that! I dont recommend ‘del’ over ‘remove’, please feel free to use the latter. I just didn’t remember whether ‘remove’ was implemented! :)

**Comment by kjetilk** [comment]
Actually, it fails if I go with `remove`:

```
  1) UserStore (integration)
       deleteUser()
         deletes a previously saved user:
     TypeError: this.backend.remove is not a function
      at Promise.resolve.then (src/user-store.js:91:241)

  2) UserStore
       deleteUser
         should call backend.del with normalized user id:
     TypeError: [Function: del] is not a spy or a call to a spy!
      at assertCanWorkWith (node_modules/sinon-chai/lib/sinon-chai.js:53:19)
      at Proxy.<anonymous> (node_modules/sinon-chai/lib/sinon-chai.js:98:13)
      at Proxy.methodWrapper (node_modules/chai/lib/chai/utils/addMethod.js:57:25)
      at Proxy.<anonymous> (node_modules/dirty-chai/lib/dirty-chai.js:184:23)
      at Proxy.overwritingMethodWrapper (node_modules/chai/lib/chai/utils/overwriteMethod.js:78:33)
      at doAsserterAsyncAndAddThen (node_modules/chai-as-promised/lib/chai-as-promised.js:289:22)
      at Proxy.<anonymous> (node_modules/chai-as-promised/lib/chai-as-promised.js:255:20)
      at Proxy.overwritingMethodWrapper (node_modules/chai/lib/chai/utils/overwriteMethod.js:78:33)
      at store.deleteUser.then (test/unit/user-store-test.js:229:60)
```

So, I suppose we'll retain `del`?

File: src/user-store.js (hunk 1 above)
Comment (RubenVerborgh) [inline src/user-store.js]: What's the `Promise.resolve().then` stuff about?
This should just be `[return] this.backend.del('users', userKey)`.

**Comment by RubenVerborgh** [comment]
Let us release a new patch version right after this PR is merged, as @kidehen needs it urgently.

File: src/user-store.js (hunk 1 above)
Comment (megoth) [reply src/user-store.js]: Yes, you're right, don't need the promise-stuff; I'll can fix this right away

File: test/integration/user-store-test.js [hunk 2]
```diff
@@ -143,4 +143,35 @@ describe('UserStore (integration)', () => {
...
+
+      let user = {
+        id: 'alice.example.com',
+        email: 'alice@example.com'
+      }
+      let password = '12345'
+
+      return store.createUser(user, password)
+        .then(() => {
+          return store.findUser(user.id)
```
Comment (dmitrizagidulin) [inline test/integration/user-store-test.js]: It's fine to leave it as-is, but there's a neat shorthand with the promise syntax -- you can say:

```js
.then(() => store.findUser(user.id))
```
and that's equivalent to {} and return, that is, equivalent to:
```js
.then(() => {
          return store.findUser(user.id)
})
```

**Comment by dmitrizagidulin** [comment]
Looks great, @kjetilk, 👍 from me!

**Pull request merged**